warehouse/query-core/src/main/java/datawave/query/jexl/functions/GroupingRequiredFilterFunctions.java

```diff
@@ -93,9 +92,12 @@
                         if (!rightSideMatches.isEmpty()) {
                             matches.addAll(rightSideMatches);
                             matches.add(match); // add the left side unmodified match
+                            tailHasMatch = true;
                         }
                     }
-                    
+                    if (!tailHasMatch) {
```

Comment:
looks like you could move this block outside of the "if Iterable else if ValueTuple" block as it is duplicated above